### PR TITLE
[Suspicious Antimalware Scan Interface DLL] Additional Query Exception for Windows Upgrades

### LINKED ISSUE #2803: [Rule Tuning] Suspicious Antimalware Scan Interface DLL
## Link to rule
https://github.com/elastic/detection-rules/blob/main/rules/windows/defense_evasion_amsi_bypass_dllhijack.toml

## Description
Miss-spell on line 112. "Distribuition" should be "Distribution" for the exception to work.

### LINKED ISSUE #2849: [Rule Tuning] Suspicious Antimalware Scan Interface DLL
## Link to rule
[defense_evasion_amsi_bypass_dllhijack](https://github.com/elastic/detection-rules/blob/main/rules/windows/defense_evasion_amsi_bypass_dllhijack.toml)

## Description
This rule monitors for file creation, and any updates to amsi.dll via Windows Update are triggering false positives. Two additions were made to the query in #2672 for similar events. Here is a new path that was falsely alerted just today:

`C:\\$WINDOWS.~BT\\Work\\C2E1A0D6-B85B-4584-B464-B34D538373E0\\amd64_microsoft-antimalware-scan-interface_31bf3856ad364e35_10.0.19041.2075_none_cb90e64f8c34137b\\f\\amsi.dll`

## Example Data
```
{
    "agent": {
      "id": "12345678-1d04-4144-ae8b-987654321fed",
      "type": "endpoint",
      "version": "8.6.2"
    },
    "process": {
      "Ext": {
        "ancestry": [
          "NzhkYjI4ZTYtMWQwNC00MTQ0LWFlOGItY2IwNmQzNzgwYjc0LTEwNDAtMTY4NjMyMzY3NC44NTY3NjM2MDA=",
          "NzhkYjI4ZTYtMWQwNC00MTQ0LWFlOGItY2IwNmQzNzgwYjc0LTE3NDgwLTE2ODYzMjM2NzQuMjcyMjYyNjAw",
          "NzhkYjI4ZTYtMWQwNC00MTQ0LWFlOGItY2IwNmQzNzgwYjc0LTEyMTkyLTE2ODYzMjM2NzQuNzUwOTk3MDA=",
          "NzhkYjI4ZTYtMWQwNC00MTQ0LWFlOGItY2IwNmQzNzgwYjc0LTUwNDQtMTY4NjMyMjA1MC4xMzk0Nzk2MDA=",
          "NzhkYjI4ZTYtMWQwNC00MTQ0LWFlOGItY2IwNmQzNzgwYjc0LTcwNC0xNjg2MjQ4MzEzLjMyMTM2NTAw",
          "NzhkYjI4ZTYtMWQwNC00MTQ0LWFlOGItY2IwNmQzNzgwYjc0LTU2MC0xNjg2MjQ4MzEyLjkwMzYwNjkwMA=="
        ],
        "code_signature": [
          {
            "trusted": true,
            "subject_name": "Microsoft Windows",
            "exists": true,
            "status": "trusted"
          }
        ]
      },
      "parent": {
        "pid": 1040
      },
      "code_signature": {
        "trusted": true,
        "subject_name": "Microsoft Windows",
        "exists": true,
        "status": "trusted"
      },
      "name": "DismHost.exe",
      "pid": 12620,
      "thread": {
        "id": 13808
      },
      "entity_id": "NzhkYjI4ZTYtMWQwNC00MTQ0LWFlOGItY2IwNmQzNzgwYjc0LTEyNjIwLTE2ODYzMjY0OTQuNDA0NTU3NDAw",
      "executable": "C:\\$WINDOWS.~BT\\Work\\F052187D-6154-43BB-9EEE-3E3F76A7C634\\DismHost.exe"
    },
    "message": "Endpoint file event",
    "@timestamp": "2023-06-09T16:04:05.8989154Z",
    "file": {
      "Ext": {
        "header_data": [],
        "entropy": 7.755565110558008,
        "header_bytes": "4dcb23935041333080d679a23d79d901",
        "windows": {
          "zone_identifier": -1
        },
        "monotonic_id": 140303
      },
      "path": "C:\\$WINDOWS.~BT\\Work\\C2E1A0D6-B85B-4584-B464-B34D538373E0\\amd64_microsoft-antimalware-scan-interface_31bf3856ad364e35_10.0.19041.2075_none_cb90e64f8c34137b\\f\\amsi.dll",
      "extension": "dll",
      "size": 20369,
      "name": "amsi.dll"
    },
    "ecs": {
      "version": "1.11.0"
    },
    "related": {
      "hosts": [
        "Computer1"
      ],
      "ip": [
        "10.10.10.10",
        "127.0.0.1",
        "::1"
      ],
      "user": [
        "system"
      ],
      "mac": [
        "01:23:45:67:89:AB"
      ]
    },
    "Effective_process": {
      "name": "setuphost.exe",
      "pid": 1040,
      "entity_id": "NzhkYjI4ZTYtMWQwNC00MTQ0LWFlOGItY2IwNmQzNzgwYjc0LTEwNDAtMTY4NjMyMzY3NC44NTY3NjM2MDA=",
      "executable": "C:\\$WINDOWS.~BT\\Sources\\setuphost.exe"
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "endpoint.events.file"
    },
    "elastic": {
      "agent": {
        "id": "12345678-1d04-4144-ae8b-987654321fed"
      }
    },
    "host": {
      "hostname": "Computer1",
      "os": {
        "Ext": {
          "variant": "Windows 10 Enterprise"
        },
        "kernel": "1909 (10.0.18363.1977)",
        "name": "Windows",
        "family": "windows",
        "type": "windows",
        "version": "1909 (10.0.18363.1977)",
        "platform": "windows",
        "full": "Windows 10 Enterprise 1909 (10.0.18363.1977)"
      },
      "ip": [
        "10.10.10.10",
        "127.0.0.1",
        "::1"
      ],
      "name": "Computer1",
      "id": "98765432-1bf0-40b3-b767-abcdef123456",
      "mac": [
        "01:23:45:67:89:AB"
      ],
      "architecture": "x86_64"
    },
    "event": {
      "agent_id_status": "verified",
      "sequence": 889692,
      "ingested": "2023-06-09T16:04:30Z",
      "created": "2023-06-09T16:04:05.8989154Z",
      "kind": "event",
      "module": "endpoint",
      "action": "creation",
      "id": "N67UGn29vKcqKXB4++++9+xW",
      "category": [
        "file"
      ],
      "type": [
        "creation"
      ],
      "dataset": "endpoint.events.file"
    },
    "user": {
      "domain": "NT AUTHORITY",
      "name": "system",
      "id": "S-1-5-18"
    }
  }
```

### PR DESCRIPTION
**Pull request by MakoWish**
## Issues

- Closes #2849 
- Closes #2803 
- Relates #2672 

## Summary

This rule monitors for file creation/modification, and any updates to amsi.dll via Windows Update, or OS updates, are triggering false-positives. This PR adds the following path as an exception:

`?:\$WINDOWS.~BT\Work\*\*`

## Contributor checklist

- [X] Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- [X] Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by MakoWish** [comment]
Pinging @w0rk3r

**Comment by MakoWish** [comment]
Pinging @terrancedejesus @w0rk3r

File: rules/windows/defense_evasion_amsi_bypass_dllhijack.toml [hunk 1]
```diff
@@ -108,7 +108,7 @@ type = "eql"
 
 query = '''
 file where host.os.type == "windows" and event.action != "deletion" and file.path != null and
- file.name : ("amsi.dll", "amsi") and not file.path : ("?:\\Windows\\system32\\amsi.dll", "?:\\Windows\\Syswow64\\amsi.dll", "?:\\$WINDOWS.~BT\\NewOS\\Windows\\WinSXS\\*", "?:\\Windows\\SoftwareDistribuition\\Download\\*")
+ file.name : ("amsi.dll", "amsi") and not file.path : ("?:\\Windows\\system32\\amsi.dll", "?:\\Windows\\Syswow64\\amsi.dll", "?:\\$WINDOWS.~BT\\NewOS\\Windows\\WinSXS\\*", "?:\\$WINDOWS.~BT\\Work\\*\\*", "?:\\Windows\\SoftwareDistribuition\\Download\\*")
```
Comment (w0rk3r) [inline rules/windows/defense_evasion_amsi_bypass_dllhijack.toml:111]:
```suggestion
 file.name : ("amsi.dll", "amsi") and not file.path : ("?:\\Windows\\system32\\amsi.dll", "?:\\Windows\\Syswow64\\amsi.dll", "?:\\$WINDOWS.~BT\\NewOS\\Windows\\WinSXS\\*", "?:\\$WINDOWS.~BT\\Work\\*\\*", "?:\\Windows\\SoftwareDistribution\\Download\\*")
```

Resolves a typo (https://github.com/elastic/detection-rules/issues/2803)

**Review by w0rk3r — APPROVED** [review]
LGTM after we bump updated date

**Comment by MakoWish** [comment]
Bumped

**Review by imays11 — APPROVED** [review]
Looks good!

**Comment by MakoWish** [comment]
I can't merge, so would appreciate if someone could.

**Pull request merged**